lib/proto.js

```diff
@@ -148,15 +148,15 @@ function write (_obj, msg, num) {
   if (_obj === undefined || _obj === null) {
     obj = mixin ? mixin() : {}
   } else if (_obj instanceof Error) {
-    const serializers = this[serializersSym]
-    obj = Object.assign(mixin ? mixin() : {}, serializers.err(_obj))
-    // backward compat, set message
-    obj.message = undefined
+    obj = Object.assign(mixin ? mixin() : {}, { err: _obj })
     if (!msg) {
       msg = _obj.message
     }
   } else {
     obj = Object.assign(mixin ? mixin() : {}, _obj)
+    if (!msg && _obj.err) {
```

Comment:
It seems like `obj = { err: { foo: 'bar' } }` would make it to this branch. Is that correct? If so, is it still acceptable for `msg = undefined`?

Reply:
@mihai1voicescu ping

Reply:
To reach this branch it should already be set to `undefined` so setting it again to `undefined ` should be OK (I could also check if the message property exists but this should be a rare occurrence).

Also maybe we should check that `msg === undefined` instead of `!msg` because someone may want to pass values like boolean `false` or number `0` for some reason.

What do you guys think?

Reply:
What I'm saying is that you're detecting in `log.info({ err: {} }, 'a message')` if the `'a message'` string is `undefined` and then assuming the object assigned to `err` has a `message` property. What happens with that assumption?

Reply:
If the assumption (`err has a message property`) proves false, then the msg (`'a message'`) will be set from `undefined` to `undefined`, which should be OK.

Sorry for the late responses, having some problems with the notifications 😞

Reply:
Can we cover that case in the tests?

Reply:
Sure.

Should I also swap to checking `msg === undefined` instead of `!msg`? Seems more correct.

Reply:
You can try it. If the tests pass without having to change them, I'd accept it. But if they don't, then I'd save that for another PR that targets v7.

Reply:
Both these changes

> Merge the 2 error flows:
> 
> if obj is of type Error wrap it in { err }
> if msg not present set to error message (if present)

should imply a major API change(v7)

The tests have already changed, because when passing an Error object it now gets transformed to `{ err }`.

The old behaviour was to set the `msg` property to `Error.message` if it was missing (`undefined`) and obj is of type Error. Now the 2 flows merge, this means they can either both default to `Error.message` or none will, which both imply modifying the tests and the behaviour. The correct choice, imho, is to default it to Error.message.

**If I understood correctly you just want to split these 2 changes into 2 PRs**, but both of them imply major API changes, altering the tests and are linked together. Also, splitting them implies making test changes in the first PR just so I can throw them away in the next PR, which seems wasteful.

Reply:
Oh, I forgot this is already targeting v7. Proceed with the change.